### PR TITLE
feat: add no_std support for ed448-goldilocks-plus

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
@mikelodder7 could you please look and publish version `0.13.2` to `crates.io` if everything is ok? Turns out this crate doesn't support no_std as I previously thought.
```
cargo build -F zeroize --no-default-features --target thumbv6m-none-eabi

   Compiling cfg-if v1.0.0
   Compiling getrandom v0.2.15
error[E0463]: can't find crate for `std`
 --> /.cargo/registry/src/index.crates.io-6f17d22bba15001f/getrandom-0.2.15/src/error_impls.rs:1:1
  |
1 | extern crate std;
  | ^^^^^^^^^^^^^^^^^ can't find crate
```

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -45,8 +45,6 @@ extern crate std;
 
 #[cfg(all(feature = "alloc", not(feature = "std")))]
 use alloc::{boxed::Box, string::ToString, vec::Vec};
-#[cfg(feature = "std")]
-use std::{boxed::Box, string::ToString, vec::Vec};
```
Comment (StackOverflowExcept1on) [inline src/lib.rs:49]: bcz [`std::prelude`](https://doc.rust-lang.org/std/prelude/index.html) has `Box`, `ToString`, `Vec`

File: src/decaf/points.rs [hunk 2]
```diff
@@ -492,6 +495,7 @@ impl TryFrom<Box<[u8]>> for CompressedDecaf {
     }
 }
 
+#[cfg(any(feature = "alloc", feature = "std"))]
```
Comment (mikelodder7) [inline src/decaf/points.rs:498]: Why is this necessary? AFAIK, no allocations occur in this code block?

File: src/decaf/points.rs [hunk 3]
```diff
@@ -503,6 +507,7 @@ impl TryFrom<DecafPointBytes> for CompressedDecaf {
...
```
Comment (mikelodder7) [inline src/decaf/points.rs:510]: Same as above

File: src/lib.rs (hunk 1 above)
Comment (mikelodder7) [reply src/lib.rs:49]: But again, in no_std, these are not included (usually) be default.

File: src/decaf/points.rs (hunk 2 above)
Comment (StackOverflowExcept1on) [reply src/decaf/points.rs:498]: compiler can't find String type here

File: src/lib.rs (hunk 1 above)
Comment (StackOverflowExcept1on) [reply src/lib.rs:49]: Yes, but if built with `--all-features`, it creates warning message

File: src/decaf/points.rs (hunk 2 above)
Comment (StackOverflowExcept1on) [reply src/decaf/points.rs:498]: no allocations, but also no `extern alloc` and the compiler cannot find `String`

**Comment by StackOverflowExcept1on** [comment]
Thank you, for the speed of the review! I also see this error on my CI but can't reproduce it:
```
error[E0432]: unresolved import `elliptic_curve::bigint::U704`
 --> /home/runner/.cargo/git/checkouts/ed448-goldilocks-e49e0e1c5eff34cb/91cdbe6/src/field/element.rs:7:34
  |
7 |         Encoding, NonZero, U448, U704,
  |                                  ^^^^ no `U704` in the root
  |
  = help: consider importing one of these type aliases instead:
          crate::elliptic_curve::consts::U704
          elliptic_curve::consts::U704
```
https://github.com/StackOverflowExcept1on/frost/actions/runs/12397926181/job/34609447922

**Comment by StackOverflowExcept1on** [comment]
@mikelodder7 I also think it makes sense to enable github actions in this repository here: https://github.com/mikelodder7/Ed448-Goldilocks/actions. It's free, but disabled for forks by default.
